lib/instrumentation/transaction.js

```diff
@@ -161,6 +161,23 @@ Transaction.prototype.toJSON = function () {
     }
   }
 
+  // add sample_rate to transaction
```

Comment:
This is the edge case handling for places where we're not just passing on the sample rate. (unsampled MUST have a sample_rate of 0, invalid/empty trace state, etc.).  Tests for these edge cases start here: https://github.com/elastic/apm-agent-nodejs/pull/2038/files#diff-800eab1f0e0cb3be24e7db4feea6cd572dc4ccbce5caf260923b1828ea3ff290R47